src-cpp/ConsumerImpl.cpp

```diff
@@ -58,6 +58,8 @@ RdKafka::Consumer *RdKafka::Consumer::create (RdKafka::Conf *conf,
   if (!(rk = rd_kafka_new(RD_KAFKA_CONSUMER, rk_conf,
                           errbuf, sizeof(errbuf)))) {
     errstr = errbuf;
+    // rd_kafka_new() takes ownership only if succeeds
+    rd_kafka_conf_destroy(rk_conf);
```

Comment:
Sorry forgot to comment on this last time around, rk_conf may be NULL for the Producer and simple Consumer, so need and if-guard:

```c
if (rk_conf)
  rd_kafka_conf_destroy(rk_conf);
```

Reply:
Good catch. I didn't notice.